react/components/atoms/lightbox/lightbox.js

```diff
@@ -69,10 +137,161 @@ export class Lightbox extends PureComponent {
+                    duration: this.props.translateAnimationDuration,
+                    useNativeDriver: true
+                })
+            ]).start(() => {
+                const scaleRatio = this.baseScale._value * this.fullZoomedInValue;
+                this.lastScale = scaleRatio;
+                this.baseScale.setValue(this.lastScale);
+                this.scaleRate.setValue(1);
+                this.setState({ zooming: false, zoomed: true });
+            });
+        });
+    };
+
+    _doubleTapZoomOut = () => {
+        this.setState({ zooming: true }, () => {
+            Animated.parallel([
+                Animated.timing(this.scaleRate, {
+                    toValue: this.fullZoomedOutValue,
+                    duration: this.props.zoomAnimationDuration,
+                    useNativeDriver: true
+                }),
+                Animated.timing(this.baseScale, {
+                    toValue: 1,
+                    duration: this.props.zoomAnimationDuration,
+                    useNativeDriver: true
+                }),
+                Animated.timing(this.state.translateX, {
+                    toValue: 0,
+                    duration: this.props.zoomAnimationDuration,
+                    useNativeDriver: true
+                }),
+                Animated.timing(this.state.translateY, {
+                    toValue: 0,
+                    duration: this.props.zoomAnimationDuration,
+                    useNativeDriver: true
+                })
+            ]).start(() => {
+                this.lastScale = 1;
+                this.baseScale.setValue(this.lastScale);
+                this.setState({ zooming: false, zoomed: false });
+            });
+        });
+    };
+
     _imageSource = () => {
         return this.props.uri ? { uri: this.props.uri } : this.props.src;
     };
 
+    /**
+     * gets the Y coordinate to zoom in to.
+     *
+     * @param {Object} event The double tap event
+     */
+    _getDoubleTapZoomInTranslateY = event => {
+        const y = event.y;
+        const middle = this.screenHeight / this.fullZoomedInValue;
+        const touchMiddleDistance = middle - y;
+
+        // if the distance of the coordinate is greater than the translate treshold
+        // then assume the treshold value as the X coordinate to zoom in to
+        const translateY = Math.min(this.translatedYTreshold, Math.abs(touchMiddleDistance));
+        return y > middle ? -1 * translateY : translateY;
+    };
+
+    /**
+     * gets the X coordinate to zoom in to.
+     *
+     * @param {Object} event The double tap event
+     */
+    _getDoubleTapZoomInTranslateX = event => {
+        const x = event.x;
+        const middle = this.screenWidth / this.fullZoomedInValue;
+        const touchMiddleDistance = x - middle;
+
+        // if the distance of the coordinate is greater than the translate treshold
+        // then assume the treshold value as the X coordinate to zoom in to

```

Comment:
with the comments of the function, either maintain this one and remove the function one or vice-versa